### ISSUE TITLE
No sound on Debian linux/amd64

### ISSUE THREAD

**Issue by kostix** [sergiou87/open-supaplex#22]
I have built opensupaplex from a4dc47edcf76dda524f383ddfcb22984a3549382, and I have failed to have any sound on an up-to-date Debian Bullseye system, linux/amd64.

The game starts normally and says it had managed to initialize audio:

```
supaplex$ ./opensupaplex -d
INFO: Debug mode enabled
INFO: Logging system initialized.
INFO: SDL_Init succeeded.
INFO: Trying to initialize audio: 44100Hz, 2 channels, format 0x8010, 512 bytes buffer
INFO: Audio initialized: 44100Hz, 2 channels, format 0x8010, 512 bytes buffer
```
but then I hear neither sound effects nor music. The volumes for both are at 10.

I have tried to use different audio backends via the `SDL_AUDIODRIVER` env. variable;
I have tried to set it to `pulse` (the default on my system), `alsa`, and `dsp`.
With `pulse` and `alsa` (and w/o overriding the standard logic of the backend selection) I see an audio stream for opensupaplex in the list of active audio streams (using `pavucontrol`), and I can verify that the selection works: when I select the `alsa` backend the stream is explicily displayed as alsa-driven by `pavucontrol`. Selecting the `dsp` backend makes `pavucontrol` display no audio stream for opensupaplex.

All the other sound-producing apps work just OK so I have hard time blaming the system config.

Please find attached:
* [opensupapled-ldd.txt](https://github.com/sergiou87/open-supaplex/files/7009191/opensupapled-ldd.txt) -- the output of running `ldd` on the generated executable; it lists no missing libs, and the set looks sensible.
* The output of `strace` from a typical run [strace.log.gz](https://github.com/sergiou87/open-supaplex/files/7009194/strace.log.gz). The audio backend for SDL is not explicitly overridden in this run.
* The output of running `ls -1a` in the directory with Supaplex: [ls.txt](https://github.com/sergiou87/open-supaplex/files/7009204/ls.txt). Here are the files from the original `supaplex.zip` plus the contents of the `resources` directory from the OpenSupaplex source code.

I'd be happy to debug it on my system, if it would be useful for you (to be honest, using the `-d` command-line option did not do much difference compared to a normal run.)

**Comment by sergiou87** [comment]
Weird! I wouldn't expect anything wrong then 🤔 

Looking at `ls -la` seems to have the `audio` folder, which I assume has files in it ([these](https://github.com/sergiou87/open-supaplex/tree/master/resources/audio), more specifically). Does the rest of the game work fine?

In the menu, when you press <kbd>Esc</kbd>, you can change the volume of both music and effects. And I can't recall what the defaults are, but of course make sure music/sound fx are enabled in the settings screen (sorry about these dumb suggestions, just making sure everything is correctly configured in that fron 😅 ).

I know some people have built it for other Linux platforms (like Raspbian or Gentoo), so I'd be surprise if it didn't work on Debian 🤔

**Comment by kostix** [comment]
The files are in place:
```
.
..
base-adlib.wav
base-beep.wav
base-blaster.wav
base-sample.wav
bug-adlib.wav
bug-beep.wav
bug-blaster.wav
bug-sample.wav
exit-adlib.wav
exit-beep.wav
exit-blaster.wav
exit-sample.wav
explosion-adlib.wav
explosion-beep.wav
explosion-blaster.wav
explosion-sample.wav
fall-adlib.wav
fall-beep.wav
fall-blaster.wav
fall-sample.wav
infotron-adlib.wav
infotron-beep.wav
infotron-blaster.wav
infotron-sample.wav
music-beep.xm
music-blaster.xm
music-roland.xm
push-adlib.wav
push-beep.wav
push-blaster.wav
push-sample.wav
```

Actually, I wouldn't know I need them (because the README does not hint about that, I think, so I thought I only needed the original game + the compiled binary), but w/o the "audio" directory, at startup, `opensupaplex` would tell me it failed to open `audio/some-file.xm` so I searched a bit and made sure I had the files in place, and the warning gone away.

So it _looks_ like it actually tries to play something :-)

**Comment by sergiou87** [comment]
Yeah, for music you might need either libmikmod or libmodplug installed, since one of those is what SDL_mixer uses to play music. However, wav files should play anyway 🤔

**Comment by kostix** [comment]
Just did a clean build of a4dc47e, and copied (from the Debian Bullseye Docker contained I did the build in) just the necessary files using `tar -cf - -C resources . -C ../linux opensupaplex | gzip -c >/opt/opensupaplex.tar.gz`, then copied it to the host system using `docker cp`, unwrapped, and run—it all works now: I have both music and sounds :laughing: 

I have no idea what's the problem, really.
I have just veried the individual WAV files under `audio` are OK.

May be it's not possible to mix the contents of the stock `supaplex.zip` and "the new" combination of the stuff from OpenSupaplex's "resources"?

Anyway, I'm closing the issue. Sorry for confusion (and kudos for cherishing my old memories :+1: ).
